pineappl_capi/src/lib.rs

```diff
@@ -1357,3 +1355,253 @@ pub unsafe extern "C" fn pineappl_string_delete(string: *mut c_char) {
         mem::drop(unsafe { CString::from_raw(string) });
     }
 }
+
+// Here starts the generalized C-API interface.
+
+/// Type for defining the interpolation object
+#[repr(C)]
+pub struct InterpTuples {
+    node_min: f64,
+    node_max: f64,
+    nb_nodes: usize,
+    interp_degree: usize,
+    reweighting_method: *const c_char,
```

Comment:
For the three members that are `*const c_char` it would nice if we could use directly or indirectly use the `enum`s from Rust and use them in C. Maybe a
```rust
pub type InterpMeth = pineappl::interpolation::InterpMeth;
```
works? If `#[repr(C)]` is needed we can add it in the Rust crate.

Reply:
This is working perfect for `InterpTuples` - and the code is much cleaner. Now, trying to do the same for the others.

Reply:
Addressed in [016eb4b](https://github.com/NNPDF/pineappl/pull/299/commits/016eb4bba61fb40d1fa3c0cc272d59667c499ffe). I wanted to promote this:
https://github.com/NNPDF/pineappl/blob/016eb4bba61fb40d1fa3c0cc272d59667c499ffe/pineappl_capi/src/lib.rs#L1462

to `ScaleFuncForm` but the enumerator `Scale` clashes with `Kinematic::Scale`.

PS: Sorry I re-introduced the wrong output when doing `git pull --rebase`.